blocks/search-results/search-results.js

```diff
@@ -0,0 +1,231 @@
+      });
+    }
+    paginationItems.append(firstPageArrow);
+
+    let currentPage = 0;
+    while (currentPage < count / pageSize) {
+      const li = document.createElement('li');
+      const span = document.createElement('span');
+      span.textContent = (currentPage + 1).toString();
+      if (currentPage === page) {
+        li.classList.add('active-page');
+      } else {
+        const pageNumber = currentPage;
+        span.addEventListener('click', () => {
+          search(indexJson, query, pageNumber, container, paginationArrows, curLocation, pageSize);
+        });
+      }
+      li.append(span);
+      paginationItems.append(li);
+      currentPage += 1;
+    }
+
+    const lastPageArrow = document.createElement('li');
+    lastPageArrow.classList.add('nav-arrow', 'last');
+    if (page === pages - 1) {
+      lastPageArrow.classList.add('nav-disabled');
+    } else {
+      const arrowImage = document.createElement('img');
+      [, arrowImage.src] = paginationArrows;
+      lastPageArrow.append(arrowImage);
+      lastPageArrow.addEventListener('click', () => {
+        search(indexJson, query, pages - 1, container, paginationArrows, curLocation, pageSize);
+      });
+    }
+    paginationItems.append(lastPageArrow);
+  }
+}
+
+function buildSearchInputField(value, placeholder) {
+  const searchInput = document.createElement('input');
+  searchInput.id = 'search';
+  searchInput.name = 'q';
+  searchInput.maxLength = 50;
+  searchInput.placeholder = placeholder;
+  if (value) {
+    searchInput.value = value;
+  }
+  return searchInput;
+}
+
+function buildSearchButton(buttonContent) {
+  const searchButton = document.createElement('button');
+  const buttonSpan = document.createElement('span');
+  searchButton.append(buttonSpan);
+  buttonSpan.append(buttonContent.textContent.trim());
+  buttonSpan.append(buttonContent.querySelector('picture'));
+  buttonContent.textContent = '';
+  searchButton.type = 'button';
+  return searchButton;
+}
+
+function buildSearchSection(block) {
+  const searchSection = document.createElement('div');
+  searchSection.classList.add('search');
+  block.append(searchSection);
+  return searchSection;
+}
+
+export default async function decorate(block, curLocation = window.location) {
+  const indexJson = await fetchData('/query-index');
+
+  const queryParams = new URLSearchParams(curLocation.search);
+  const query = queryParams.get('q')?.trim();
+  const page = Number(queryParams.get('p'));
+  const s = queryParams.get('s');
+  const pageSize = s ? Number(s) : 10;
+  const buttonContent = block.querySelector('div.search-results > div > div');
+  buttonContent.closest('div').remove();
+  const paginationArrows = block.querySelectorAll('div.search-results > div > div > picture > img');
+  const paginationArrowsSrc = [...paginationArrows].map((arrow) => arrow.src);
+  paginationArrows.forEach((arrow) => arrow.closest('div').remove());
```

Comment:
I would not use document editing for icons because it creates more complexity for the authors and more potential to break things. Changing icons is not an author task in any case.
Let's put icons under /icons, generate some layout markup in .js and specify the icons in the .css
wdyt?

Reply:
Yeah, this was one of the questions I had about the approach: should we allow authors to change the icons or is it a developer thing? 
I'm ok with your suggestion: I will remove all content from the block in the Word document and move the icons under `/icons` in the git repo.

Reply:
Done